app/code/Magento/Store/Model/StoreCreateManagement.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Magento\Store\Model;
+
+use Magento\Store\Api\StoreCreateManagementInterface;
+
+class StoreCreateManagement implements StoreCreateManagementInterface
+{
+    /**
+     * @var \Magento\Store\Model\WebsiteFactory
+     */
+    private $websiteFactory;
+
+    /**
+     * @var \Magento\Store\Model\GroupFactory
+     */
+    private $groupFactory;
+
+    /**
+     * @var \Magento\Store\Model\StoreFactory
+     */
+    private $storeFactory;
+
+    /**
+     *
+     * @param \Magento\Store\Model\WebsiteFactory     $websiteFactory
+     * @param \Magento\Store\Model\GroupFactory       $groupFactory
+     * @param \Magento\Store\Model\StoreFactory       $storeFactory
```

Comment:
Please use the interface factoies instead of model factories to handle changing of DI preference.

Applicable to all 3 factories

I.e. `\Magento\Store\Api\Data\StoreInterfaceFactory`